skimage/measure/_marching_cubes_classic.py

```diff
@@ -179,79 +178,6 @@ def mesh_surface_area(verts, faces):
     return ((np.cross(a, b) ** 2).sum(axis=1) ** 0.5).sum() / 2.
 
 
-def correct_mesh_orientation(volume, verts, faces, spacing=(1., 1., 1.),
-                             gradient_direction='descent'):
-    """
-    Correct orientations of mesh faces.
-
-    Parameters
-    ----------
-    volume : (M, N, P) array of doubles
-        Input data volume to find isosurfaces. Will be cast to `np.float64`.
-    verts : (V, 3) array of floats
-        Array containing (x, y, z) coordinates for V unique mesh vertices.
-    faces : (F, 3) array of ints
-        List of length-3 lists of integers, referencing vertex coordinates as
-        provided in `verts`.
-    spacing : length-3 tuple of floats
-        Voxel spacing in spatial dimensions corresponding to numpy array
-        indexing dimensions (M, N, P) as in `volume`.
-    gradient_direction : string
-        Controls if the mesh was generated from an isosurface with gradient
-        descent toward objects of interest (the default), or the opposite.
-        The two options are:
-        * descent : Object was greater than exterior
-        * ascent : Exterior was greater than object
-
-    Returns
-    -------
-    faces_corrected (F, 3) array of ints
-        Corrected list of faces referencing vertex coordinates in `verts`.
-
-    Notes
-    -----
-    Certain applications and mesh processing algorithms require all faces
-    to be oriented in a consistent way. Generally, this means a normal vector
-    points "out" of the meshed shapes. This algorithm corrects the output from
-    `skimage.measure.marching_cubes_classic` by flipping the orientation of
-    mis-oriented faces.
-
-    Because marching cubes could be used to find isosurfaces either on
-    gradient descent (where the desired object has greater values than the
-    exterior) or ascent (where the desired object has lower values than the
-    exterior), the ``gradient_direction`` kwarg allows the user to inform this
-    algorithm which is correct. If the resulting mesh appears to be oriented
-    completely incorrectly, try changing this option.
-
-    The arguments expected by this function are the exact outputs from
-    `skimage.measure.marching_cubes_classic`. Only `faces` is corrected and
-    returned, as the vertices do not change; only the order in which they are
-    referenced.
-
-    This algorithm assumes ``faces`` provided are all triangles.
-
-    See Also
-    --------
-    skimage.measure.marching_cubes_classic
-    skimage.measure.mesh_surface_area
-
-    """
-    warn(DeprecationWarning("`correct_mesh_orientation` is deprecated for "
-                            "removal as `marching_cubes_classic` now "
-                            "guarantees correct mesh orientation."))
```

Comment:
Why was there no notice about when exactly this was going to get removed?

Reply:
@hmaarrfk the warning was added in https://github.com/scikit-image/scikit-image/commit/29725068ce582f9992a26a852643876e22f11bec, prior to 0.12.0 release. I think approximately around that time we worked out the convention for deprecating the functionality. Wasn't too strict back then :).